File: src/main/java/com/hospital/service/PatientService.java
package com.hospital.service;

import com.hospital.dto.request.PatientRequest;
import com.hospital.dto.response.PatientResponse;

import java.util.List;

/**
 * Business contract for Patient profile management.
 */
public interface PatientService {

    /** Create a new patient profile linked to an existing user. */
    PatientResponse createPatientProfile(Long userId, PatientRequest request);

    /** Fetch a patient by their profile ID. */
    PatientResponse getPatientById(Long id);

    /** Fetch a patient by their linked user ID. */
    PatientResponse getPatientByUserId(Long userId);

    /** List all patients (admin use). */
    List<PatientResponse> getAllPatients();

    /** Update an existing patient profile. */
    PatientResponse updatePatientProfile(Long id, PatientRequest request);

    /** Delete a patient profile by ID. */
    void deletePatient(Long id);
}

File: frontend/src/services/patientService.js
import api from './api';

export const patientService = {
  getMe:        ()             => api.get('/patients/me'),
  getHistory:   ()             => api.get('/patients/me/history'),
  getById:      (id)           => api.get(`/patients/${id}`),
  getByUserId:  (userId)       => api.get(`/patients/user/${userId}`),
  getAll:       ()             => api.get('/patients'),
  create:       (userId, data) => api.post(`/patients/user/${userId}`, data),
  update:       (id, data)     => api.put(`/patients/${id}`, data),
  delete:       (id)           => api.delete(`/patients/${id}`),
};
